### PR TITLE
Add relationships for queryRecord

### PR DESCRIPTION
**Pull request by mikaalnaik**

### CODE REVIEW

File: src/json-api/serializer.ts [hunk 1]
```diff
@@ -35,6 +35,6 @@ type jsonApiResource = {
 }
 
 type jsonApiPayload = {
```
Comment (egjiri) [inline src/json-api/serializer.ts:37]:
```js
type jsonApiPayload = jsonAPIIndexPayload | jsonAPIShowPayload

type jsonAPIIndexPayload = {
  data: jsonApiResource[],
  meta: Record<string, unknown>,
}

type jsonAPIShowPayload = {
  data: jsonApiResource,
}
```

File: src/json-api/serializer.ts [hunk 2]
```diff
@@ -6,10 +6,10 @@ export default class JSONAPISerializer extends Serializer {
       const content = payload.data.map(item => {
         return this.buildModel({ id: item.id, ...item.attributes, ...extractRelationships(item.relationships) });
       });
-      return this.proxyContent(content, payload.meta);
+      return this.proxyContent(content, (payload as jsonAPIIndexPayload).meta);
     } else {
-      const { id, attributes } = payload.data as jsonApiResource;
-      return this.buildModel({ id, ...attributes });
+      const { id, attributes, relationships } = payload.data as jsonApiResource;
```
Comment (egjiri) [inline src/json-api/serializer.ts:11]: Can this be `(payload as jsonAPIShowPayload).data instead?

File: src/json-api/serializer.test.ts [hunk 3]
```diff
@@ -15,7 +15,41 @@ test('test JSONAPISerializer', () => {
...
   };
-  const response = new Serializer(testCreateModel).normalizePayload(payload);
-  expect(response).toEqual([{ id: 1, firstName: 'John' }, { id: 2, firstName: 'Jane' }, { id: 3, firstName: 'Joe', friend: { id: 1894 }}]);
-  expect(response.meta).toEqual({ pageNumber: 1, pageSize: 25, totalPages: 1, totalCount: 2 });
+  const queryResponse = new Serializer(testCreateModel).normalizePayload(arrayPayload);
+  expect(queryResponse).toEqual([{ id: 1, firstName: 'John' }, { id: 2, firstName: 'Jane' }, { id: 3, firstName: 'Joe', friend: { id: 1894 }}]);
+  expect(queryResponse.meta).toEqual({ pageNumber: 1, pageSize: 25, totalPages: 1, totalCount: 2 });
+});
+
+test('test JSONAPISerializer Non Array Payload', () => {
```
Comment (egjiri) [inline src/json-api/serializer.test.ts:23]: Change `Non Array Payload` to `Object Payload`

File: src/json-api/serializer.test.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import Serializer from './serializer';
 import { testCreateModel } from '../serializer.test';
 
-test('test JSONAPISerializer', () => {
-  const payload = {
+test('test JSONAPISerializer Array Payload', () => {
```
Comment (egjiri) [inline src/json-api/serializer.test.ts:4]: Actually let's change this name to `Index Payload` and the variable name below to `const indexPayload`

Comment (mikaalnaik) [reply src/json-api/serializer.test.ts]: I've also renamed the other test to `Show Payload` as well to match the pattern

**Pull request merged**